### PR TITLE
[IMP] add colombian states

### PR DESCRIPTION
**Pull request by blaggacao**
## Add complete set of colombian states. @mart-e 

I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

Signed-off-by: David Arnold dar@devco.co

### CODE REVIEW

**Comment by wtaferner** [comment]
@pedrobaeza 
I like to have this information to be structured in core too, but it was denied for my country as they will only include if it is a legal requirement to have the state in the address...at least this was the answer to my pull request https://github.com/odoo/odoo/pull/13069

**Comment by pedrobaeza** [comment]
@wtaferner, thanks for the info. It seems that it is mandatory according this other PR: https://github.com/odoo/odoo/pull/14066.

**Comment by blaggacao** [comment]
Ok, thanks for the info! I advance the answer here:
The colombian states, just as the municipalities have a code assigned from the colombian national statistical institute, the DANE. We commonly refer to those codes as "codigos DANE", therefore. Now at the end of the year you need to attribute every single transaction to a partner, and attribute it to the corresponding DANE codes (all of municipality, state and country). Without this information, you cannot comply with this year's' end legal requirement. Obviously we need to overcharge the DANE code in the localisation module, which we will do with our OCA-pending `l10n_co_geo`. We would step by step be able to factor this information into the base `l10n_co` module, but I don't think it is wise to make all steps at once.

In conclusion: Yes, this is a legal requirement for Colombia. No, by it's own, this commit doesn't completely satisfy it but we plan to push things further step after step.

**Comment by blaggacao** [comment]
To give proof, is a bit more complicated but not unfeasible:
You need to download the official "Prevalidador" and try to fill in for example the partner information for the FORM 1001, then you can complete the process and export an xml file where you have the country, state and city encoded via the DANE code. If any colombian company is not able to completely register a transaction in this "Prevalidador", this transaction is fiscally not acknowledged.

http://www.dian.gov.co/contenidos/otros/prevalidadores.html

This is probably the single source of truth for many other l10n_co requirements as well and the biggest headache of the country's businesses.

**Comment by blaggacao** [comment]
Another argument, yet I deem the above ones are sufficient: there are special tax regimes in some states, like you cans see here: 
http://estatuto.co/?e=713

**Comment by JKE-be** [comment]
Thanks for the pr
@blaggacao ... It is required in the address format of Colombia ? 
If not, please add it in l10n...

**Comment by blaggacao** [comment]
@JKE-be I'm not 100% sure of your definition of "required"?
Let me put it this way:
If you don't put it, a letter would not be delivered. I think, that's "required".

**Comment by mart-e** [comment]
Was merged at 5b9015295100d7b2a9d8a92efc34bbf8613314b5 but forgot to close :)

**Comment by blaggacao** [comment]
Yeay!!! Thanks @mart-e This is very welcome traction to our cause!
